blog/2020/12/package_latency.md

```diff
@@ -0,0 +1,718 @@
+@def authors = "Tim Holy and Nathan Daly"
+@def published = "1 December 2020"
+@def title = "Reducing compiler latency in Julia packages"
+@def rss_pubdate = Date(2020, 12, 1)
+@def rss = """Julia runs fast, but suffers from latency due to compilation. This post describes how to detect and eliminate major sources of latency in packages."""
+
+\toc
+
+[The Julia programming language][Julia] delivers remarkable runtime performance and flexibility.  Julia's flexibility depends on the ability to of methods to handle arguments of many different types.  This flexibility would be in competition with runtime performance, were it not for the "trick" of *method specialization*.  Julia compiles a separate "instance" of a method for each distinct combination of argument types; this specialization allows code to be optimized to take advantage of specific features of the inputs, eliminating most of the *runtime* cost that would otherwise be the result of Julia's flexibility.
+
+Unfortunately, method specialization has its own cost: compiler latency.  Since compilation is expensive, there is a measurable delay that occurs on first invokation of a method for a specific combination of argument types.  There are cases where one can do some of this work once, in advance, using utilities like [`precompile`] or building a custom system with [PackageCompiler].  In other cases, the number of distinct argument types that a method might be passed seems effectively infinite, and in such cases precompilation seems unlikely to be a comprehensive solution.
```

Comment:
"using utilities like [`precompile`] or building a custom system with [PackageCompiler]" sounds slightly off to me. Perhaps "or building" -> "or by building" and "custom system" -> "custom system image" or so? :)